staging/src/k8s.io/cloud-provider/controllers/node/node_controller_test.go

```diff
@@ -764,6 +764,7 @@ func Test_syncNode(t *testing.T) {
 							Effect: v1.TaintEffectNoSchedule,
 						},
 					},
+					ProviderID: "fake://12345",
```

Comment:
I think we still want a test case where the existing node doesn't have provider ID already set. I think this requires a larger refactor in the fake cloud provider though, so I'm okay to address this in a follow-up PR